src/detext/train/train.py

```diff
@@ -298,6 +303,16 @@ def model_fn(features, labels, mode, params):
 
     group_size_field = features['group_size'] if mode != tf.estimator.ModeKeys.PREDICT else None
 
+    # For multitask training
+    task_id_field = features.get('task_id', None)
+
+    # Override the default weight for multitask from input params task_ids and task_weights
+    if params.task_ids is not None:
+        weight = tf.zeros(shape=tf.shape(weight), dtype="float32")
+        for task_id, task_weight in params.task_ids.items():
+            task_mask = tf.cast(tf.equal(task_id_field, task_id), dtype=tf.float32)
+            weight += task_weight * task_mask
+
```

Comment:
weight and task_weight are orthogonal. Multiplying instead of overriding them is more appropriate (consider the case where some samples for a task are downsampled)

Reply:
It makes sense to separate them conceptually. In practice, they may not be differentiable. For example, the sample weights are equivalent to the task weights when the loss is calculated as the weighted avg of different records/tasks. But I agree it'll be flexible to separate them here for flexibility. I'll update it.